elasticsearch/class-elasticsearch.php

```diff
@@ -24,13 +25,35 @@ protected function load_dependencies() {
 		}
 	}
 
-	protected function add_hooks() {
-		// Add filters and action hooks here
+	public function setup_constants() {
+		// Ensure we limit bulk indexing chunk size to a reasonable number (no limit by default)
+		if ( ! defined( 'EP_SYNC_CHUNK_LIMIT' ) ) {
+			define( 'EP_SYNC_CHUNK_LIMIT', 250 );
+		}
+	}
+
+	public function setup_hooks() {
+		add_filter( 'ep_index_name', [ $this, 'filter__ep_index_name' ], PHP_INT_MAX, 3 ); // We want to enforce the naming, so run this really late.
 	}
 
-	protected function load_commands() {
+	public function load_commands() {
 		if ( defined( 'WP_CLI' ) && WP_CLI ) {
 			WP_CLI::add_command( 'vip-es health', __NAMESPACE__ . '\Health_Command' );
```

Comment:
Considering this is unlikely to work, without the `commands/class-health-command.php` having been required first, why is the require_once in a separate method?

And why are they both `public` methods, when the public `init` method already seems to handle this?

Reply:
> why is the require_once in a separate method?

We have a `load_dependencies()` method which is intended to gather up all the code needed for the plugin to work - it's a good point about this not working if it's called independently of `load_dependencies()`.

> And why are they both public methods

I switched them all to `public` for unit tests, because I wanted to test parts of it in isolation. Technically this one doesn't have to be `public` since there isn't a test for it.

Reply:
[Using Reflection for testing private methods](https://42coders.com/how-to-use-reflection-to-test-private-and-protected-methods/) is typically preferred, rather than exposing methods that otherwise wouldn't be exposed if it weren't for unit testing.

Reply:
Ah much better, thanks @GaryJones !